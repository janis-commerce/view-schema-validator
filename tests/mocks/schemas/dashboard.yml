service: serviceName
name: example-dashboard
root: Dashboard
filters:
  - name: filterInput
    label: someLabel
    component: Input
    componentAttributes:
      icon: iconName
graphs:
  - component: Table
    name: graphName
    title: someTitleForGraph
    source:
      service: serviceName
      namespace: namespaceName
      method: methodName
      resolve: false
    endpointParameters:
      - name: status
        target: path
        value:
          dynamic: id
      - name: status
        target: query
        value:
          static: 1
    x: 0
    y: 0
    width: 6
    height: 3

  - component: BarChart
    name: graphName
    title: someTitleForGraph
    translateTitle: true
    subtitle: someSubtitleForGraph
    translateSubtitle: true
    source:
      service: serviceName
      namespace: namespaceName
      method: methodName
      resolve: false
    componentAttributes:
      options:
        width: 100%
    x: 6
    y: 0
    width: 6
    height: 3


