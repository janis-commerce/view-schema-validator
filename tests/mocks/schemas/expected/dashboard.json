{
    "service": "serviceName",
    "name": "example-dashboard",
    "root": "Dashboard",
    "filters": [
        {
            "name": "filterInput",
            "label": "someLabel",
            "component": "Input",
            "componentAttributes": {
                "icon": "iconName"
            }
        }
    ],
    "graphs": [
        {
            "component": "Table",
            "filters": [],
            "name": "graphName",
            "title": "someTitleForGraph",
            "source": {
                "service": "serviceName",
                "namespace": "namespaceName",
                "method": "methodName",
                "resolve": false
            },
            "endpointParameters": [
                {
                    "name": "status",
                    "target": "path",
                    "value": {
                        "dynamic": "id"
                    }
                },
                {
                    "name": "status",
                    "target": "query",
                    "value": {
                        "static": 1
                    }
                }
            ],
            "x": 0,
            "y": 0,
            "width": 6,
            "height": 3
        },
        {
            "component": "BarChart",
            "filters": [],
            "name": "graphName",
            "title": "someTitleForGraph",
            "translateTitle": true,
            "subtitle": "someSubtitleForGraph",
            "source": {
                "service": "serviceName",
                "namespace": "namespaceName",
                "method": "methodName",
                "resolve": false
            },
            "translateSubtitle": true,
            "componentAttributes": {
                "options": {
                    "width": "100%"
                }
            },
            "x": 6,
            "y": 0,
            "width": 6,
            "height": 3
        }
    ]
}